Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -443,47 +493,69 @@ def print_latest_version(info, verbose:, ambiguous_cask: false)
         info[:version][:latest]
       end
 
-      puts "#{formula_or_cask_s}: #{current_s} ==> #{latest_s}"
+      puts "#{package_or_resource_s}: #{current_s} ==> #{latest_s}"
+    end
+
+    # Prints the livecheck result for a resources for a given Formula.
+    sig { params(info: T::Array[Hash], verbose: T::Boolean).void }
+    def print_resources_info(info, verbose: false)
+      info.each do |r_info|
+        if r_info[:status] && r_info[:messages]
+          SkipConditions.print_skip_information(r_info)
+          Homebrew.failed = true
```

Comment:
The presence of `:status`/`:messages` values doesn't always indicate a failure condition. For example, these values would be set for a skipped resource but that's not a failure state. The previous use of `Homebrew.failed = true` was fine from my perspective (i.e., it mimicked the behavior in `#run_checks`) but this is setting the failure state for more than it should.

@nandahkrishna Can you elaborate on which scenarios you were thinking should be treated as a failure state with regard to resources? This may help to clarify where `Homebrew.failed = true` should be used outside of the `rescue` blocks.

Reply:
I wanted to make sure that an `Unable to get versions` for resources would be treated as a failure state. Prior to this change, we did not have an error status code if a formula's `livecheck` succeeded but a resource's failed. This was the only thing I had in mind so far about failure states for resources.

Reply:
Just a naive idea, we can use `:status` to identify if the resource was skipped or was it due to some failed state, and mark `Homebrew.failed = true` in that case. This way we won't get false status code in case of a skipped resource. @samford and @nandahkrishna: what are your thoughts on this ?

Reply:
It looks like GitHub lost both of our comments (my response and your reply to it), so I'll quote it from the email/DM:

> > However, this should arguably be handled somewhere after #resource_version is called in #run_checks
>
> Is this what you had in mind ?
>
> ```ruby
> check_for_resources = check_resources && formula_or_cask.is_a?(Formula) && formula_or_cask.resources.present?
> if check_for_resources
>   resource_version_info = formula_or_cask.resources.map do |resource|
>     res_skip_info ||= SkipConditions.skip_information(resource, verbose: verbose)
>     if res_skip_info.present?
>       res_skip_info
>     else
>       res_version_info = resource_version(
>         resource,
>         json:    json,
>         debug:   debug,
>         quiet:   quiet,
>         verbose: verbose,
>       )
>       if res_version_info.empty?
>         status_hash(resource, "error", ["Unable to get versions"], verbose: verbose)
>       else
>         res_version_info
>       end
>     end
>   end.compact_blank
> Homebrew.failed = true if resource_version_info.any? { |info| info[:status] == "error" }
> end
> ```

Yes, that's what I was thinking. From my limited testing, it appears to work as expected and would handle Nanda's use case. Feel free to push a commit and I'll take a final pass through this PR.

Reply:
I have pushed the change ✅